Modules/_queuemodule.c

```diff
@@ -249,7 +364,7 @@ _queue_SimpleQueue_get_impl(simplequeueobject *self, PyTypeObject *cls,
      * So we simply try to acquire the lock in a loop, until the condition
      * (queue non-empty) becomes true.
      */
-    while (self->lst_pos == PyList_GET_SIZE(self->lst)) {
+    while (self->buf.num_items == 0) {
```

Comment:
Consider to also add APIs for _empty_ and _num_items_, for less future churn in case we ever want to adjust the implementation.